src/dowel/csv_output.py

```diff
@@ -41,13 +41,33 @@ def record(self, data, prefix=''):
                     extrasaction='ignore')
                 self._writer.writeheader()
 
+            # new fieldnames detected, need to read in current file, modify, and write everything again
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+                # obtain extra fieldname
+                extra_field = (set(to_csv.keys()) ^ set(self._fieldnames)).pop()
```

Comment:
There can actually be multiple new fields, so this logic isn't completely correct.

Can you update this function to account for there being multiple new fields?